stdlib/test/utils/test_string_slice.mojo

```diff
@@ -145,8 +145,29 @@ fn test_slice_len() raises:
     assert_equal(1, len(slice5))
 
 
+fn test_slice_eq() raises:
+    var str1: String = "12345"
+    var str2: String = "12345"
+    var str3: StringLiteral = "12345"
+    var str4: String = "abc"
+    assert_true(str1.as_string_slice() == str1)
+    assert_true(str1.as_string_slice() == str2)
+    assert_true(str2.as_string_slice() == str2.as_string_slice())
+    assert_true(str1.as_string_slice() == str3)
+    assert_true(not str1.as_string_slice() == str4)
```

Comment:
Can we add explicit tests that invoke the `__ne__` too?  It struck me as odd seeing the asymmetry of overloads between `__eq__` and `__ne__` at first glance, for example.  Clarifying this with tests would be helpful.